lms/views/helpers/_via.py

```diff
@@ -1,8 +1,81 @@
 """Via-related view helpers."""
+import re
 from urllib.parse import parse_qsl, urlencode, urlparse
 
 __all__ = ["via_url"]
 
+# pylint: disable=too-few-public-methods
+
+
+class _ViaDoc:
+    """A doc we want to proxy with content type."""
+
+    GOOGLE_DRIVE_REGEX = re.compile(
+        r"^https://drive.google.com/uc\?id=(.*)&export=download$", re.IGNORECASE
+    )
+
+    def __init__(self, url, content_type=None):
+        self.url = url
+
+        if content_type is None and self.GOOGLE_DRIVE_REGEX.match(url):
+            content_type = "pdf"
+
+        self._content_type = content_type
+
+    @property
+    def is_pdf(self):
+        return self._content_type == "pdf"
+
+    @property
+    def is_html(self):
+        return self._content_type == "html"
+
+
+class _ViaClient:
+    """A small wrapper to make calling Via easier."""
+
+    def __init__(self, service_url, legacy_service_url, host_url, legacy_mode=False):
+        self.service_url = urlparse(service_url)
+        self.legacy_service_url = legacy_service_url
+
+        # Default via parameters
+        self.options = {
+            "via.client.openSidebar": "1",
+            "via.client.requestConfigFromFrame.origin": host_url,
+            "via.client.requestConfigFromFrame.ancestorLevel": "2",
+        }
+        self.legacy_mode = legacy_mode
+
+    def url_for(self, doc):
+        assert isinstance(doc, _ViaDoc), "Provided doc is a ViaDoc"
+
```

Comment:
```suggestion
```

I don't think we should bother with these kinds of checks in Python, just rely on Duck Typing (i.e. don't bother)